### PR TITLE
chore(frontend): migrate project-tooling to Vite and refacto logic code

### PR DESCRIPTION
**Pull request by ElodieENSTA**
Started from https://github.com/Project-OSmOSE/osmose-app/pull/165 and add a big refacto :
- Make API services for API requests
- Use context to handle Auth and data in the annotator
- add small updates
 - Redirect on the campaigns list after login
 - Simple navigation now require validation if updates has been made by the user
 - keep audio playback rate configuration after navigation

Also :
-  New Cypress tests for all pages except the Annotator.
- New package.json at the root of the project for common commands

### CODE REVIEW

**Review by rvovard — CHANGES_REQUESTED** [review]
Great and very, very impressive work, well done!

I honestly can't make a good review of all the code, everything was changed and we'll see what can be improved by developing with this new codebase.

I've made many tests and found a few bugs that you'll see in comments.

A few notes : 

- I don't thing that it's a good idea to have imports starting with `../../..`. In the future, we should set a base folder to be able to do imports like `src/....`.
- Just a cosmetic (but not only) preference that Erwan and I have: it could be better when files have a new line at the end. It helps with CLI tools, can ease some diffs... It no longer could bring bugs as it did, but it's a practice several developers have kept (and generaly enforced by code formatters).
- Old tests could be deleted if new Cypress tests opens the same components.

Thanks for this refacto!

File: frontend/src/App.tsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        </AuthenticatedRoute>
+
+        <AploseSkeleton>
+          <Switch>
+            <AuthenticatedRoute exact path='/datasets'><DatasetList/></AuthenticatedRoute>
+            <AuthenticatedRoute exact path='/annotation-campaigns'><AnnotationCampaignList/></AuthenticatedRoute>
+            <AuthenticatedRoute exact path='/create-annotation-campaign'><CreateAnnotationCampaign/></AuthenticatedRoute>
+            <AuthenticatedRoute path='/annotation_campaign/:id'><AnnotationCampaignDetail/></AuthenticatedRoute>
+            <AuthenticatedRoute path='/annotation_campaign/:id/edit'><EditAnnotationCampaign/></AuthenticatedRoute>
+            <AuthenticatedRoute path='/annotation_tasks/:id'><AnnotationTaskList/></AuthenticatedRoute>
```
[the remark this answers is not in the document]
Comment (ElodieENSTA) [reply frontend/src/App.tsx:67]: Yes, I'll add it, thanks.
I thought `exact` didn't work when there was url params, but it does

**Comment by ElodieENSTA** [comment]
Thanks @rvovard for your review ! I made the requested changes.

About your notes:

> * I don't thing that it's a good idea to have imports starting with `../../..`. In the future, we should set a base folder to be able to do imports like `src/....`.
What is the difference between this 2 path writing ? Why is `src/...` better than `../../..` ? Just trying to understand your request :)

> * Just a cosmetic (but not only) preference that Erwan and I have: it could be better when files have a new line at the end. It helps with CLI tools, can ease some diffs... It no longer could bring bugs as it did, but it's a practice several developers have kept (and generaly enforced by code formatters).
Okay, I configured my IDE to add it on save ^^  I tried to fix that in all files in my last commit.

> * Old tests could be deleted if new Cypress tests opens the same components.
Oh yes ! I didn't saw that folder when I made the new tests 🤦 I will make sure to transfer every old tests in Cypress. It will be made over the next developments

I also fixed a bug that happened when you played an audio file and then navigate to another annotation task : the update time interval wasn't resetting. The only way I found to reset it was to declare the interval within a useEffect and clear it in the return. I don't really like this solution since the interval is running all the time when we are in the editor but didn't find better.

File: frontend/src/services/auth/auth.service.tsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+              private history: History<LocationState>) {
+  }
+
+  async login(username: string, password: string): Promise<void> {
+    const response = await fetch(this.URI, {
+      method: 'POST',
+      headers: { "Content-Type": "application/json" },
+      body: JSON.stringify({ username, password }),
+      signal: this.controller.signal,
+    });
```
[the remark this answers is not in the document]
Comment (rvovard) [reply frontend/src/services/auth/auth.service.tsx]: No no as I said it was just a comment and must not be done immediately!
I agree that it is too much work, this merge request is already quite big :)

**Comment by rvovard** [comment]
Thanks @ElodieENSTA!

> What is the difference between this 2 path writing ? Why is `src/...` better than `../../..` ? Just trying to understand your request :)

The use of absolute paths might be useful from a developer point of view : if you refactor your component and move it in another folder, all relative imports will break. And it can be difficult to read those imports.
But it is quite personal, I've search about it and have not found many person advocating for this as an obvious best practice. You can read [this article](https://medium.com/thefork/escaping-relative-import-paths-hell-in-javascript-9f258baeb15e) which explains my point of view :)

> Okay, I configured my IDE to add it on save ^^ I tried to fix that in all files in my last commit.

Thanks a lot, it was not that urgent ^^

> I also fixed a bug that happened when you played an audio file and then navigate to another annotation task : the update time interval wasn't resetting. The only way I found to reset it was to declare the interval within a useEffect and clear it in the return. I don't really like this solution since the interval is running all the time when we are in the editor but didn't find better.

As seen together, I'll check if I can figure out what's happening during the next days.

**Comment by ElodieENSTA** [comment]
As seen together, just updated the audio time interval to 30FPS

**Pull request merged**